sky/cli.py

```diff
@@ -342,6 +343,48 @@ def _parse_env_var(env_var: str) -> Tuple[str, str]:
 ]
 
 
+def _complete_cluster_name(ctx: click.Context, param: click.Parameter,
+                           incomplete: str) -> List[str]:
+    """Handle shell completion for cluster names."""
+    del ctx, param  # Unused.
+    return global_user_state.get_cluster_names(incomplete)
+
+
+def _install_shell_completion(ctx: click.Context, param: click.Parameter,
+                              value: str):
+    """A callback for installing shell completion for click."""
+    del param  # Unused.
+    if not value or ctx.resilient_parsing:
+        return
+
+    if value == 'bash':
+        cmd = '_SKY_COMPLETE=bash_source sky > ~/.sky-complete.bash && \
+                echo "# For SkyPilot shell completion" >> ~/.bashrc && \
+                echo ". ~/.sky-complete.bash\n" >> ~/.bashrc'
```

Comment:
Can we make this idempotent? Otherwise, IIUC, if the user run the `sky --install-shell-completioin` will create multiple same lines in the `bashrc` file. Something similar to the trick in the following line can be applied.
https://github.com/skypilot-org/skypilot/blob/10967e6c17bee50827dfc6d1ea5cdd7fff2d6c8c/sky/templates/aws-ray.yml.j2#L95

Same for L366 and L370.